py17track/profile.py

```diff
@@ -108,3 +109,70 @@ async def summary(self, show_archived: bool = False) -> dict:
         for kind in summary_resp.get("Json", {}).get("eitem", []):
             results[PACKAGE_STATUS_MAP[kind["e"]]] = kind["ec"]
         return results
+
+    async def add_package(
+        self, tracking_number: str, friendly_name: Optional[str] = None
+    ) -> bool:
+        """Add a package by tracking number to the tracking list."""
+        add_resp: dict = await self._request(
+            "post",
+            API_URL_BUYER,
+            json={
+                "version": "1.0",
+                "method": "AddTrackNo",
+                "param": {"TrackNos": [tracking_number]},
+            },
+        )
+
+        _LOGGER.debug("Add package response: %s", add_resp)
+
+        if add_resp.get("Code") != 0:
+            return False
```

Comment:
We shouldn't swallow an unsuccessful response here. If we could get an `HTTP 2xx` response with a non-0 `Code`, we should raise a `RequestError`.

Reply:
This would eliminate the need for returning a `bool`, as well – if an exception isn't raised, we can assume the request was successful.